idea/testData/quickfix/expressions/removeUselessIsCheckInWhen.kt

```diff
@@ -0,0 +1,6 @@
+// "Remove useless is check" "false"
+fun foo(a: String) {
+    when (1) {
+        <caret>is Int -> { }
```

Comment:
I would like to try by myself :) But I have bad feeling. For `if` case we are modifying only condition. Like
```diff
-if (1 is Boolean) { print() }
+if (false) { print() }
```
but not whole branch:
```diff
-if (1 is Boolean) { print() }
+
```

What are you asking is to change whole `when`. It could be optimized totally:
```diff
-when (1) {
-  is Boolean -> print()
-}
+
```
and 
```diff
-when (1) {
-  is Int -> print()
-}
+ print()
```